### PR TITLE
feat(backend): read Netlify x-country header

### LINKED ISSUE #1023: Support Netlify geolocation signals
## Problem

c15t already understands first-party geo signals from Cloudflare, Vercel, and CloudFront. Netlify is a common host and already knows the visitor's country (and often subdivision), but we do not treat those signals as first-class input. Sites on Netlify therefore tend to resolve as an unknown location and hit the fallback policy even though the platform has already geolocated the request.

### Official: `context.geo`

On both serverless Functions and Edge Functions, Netlify attaches a `Context` object with:

| Field | Contents |
| --- | --- |
| `context.geo.country.code` | ISO 3166-1 alpha-2 (`DE`, `US`, …) |
| `context.geo.country.name` | Country name |
| `context.geo.subdivision.code` / `.name` | ISO 3166-2 subdivision |
| `city`, `latitude`, `longitude`, `timezone`, `postalCode` | Extra location data |

This is the documented API. Edge Functions specifically do **not** add geo as request headers — the docs say to use `context` instead.

Subdivision is the part that matters beyond country-level GDPR/UK mapping. California (CCPA) and Quebec (Law 25) need country **and** region; country-only support would still leave those visitors looking like generic `US` / `CA`.

### Function request header: `x-country`

Serverless function requests still receive `x-country` with the same 2-letter country code. This is the simple header analogue of `cf-ipcountry` / `x-vercel-ip-country`. It is **not** the same header as `x-country-code`.

There is no equivalent simple region header — `x-country` is country only.

### Internal: `x-nf-geo`

A base64-encoded JSON blob (country + subdivision + city, etc.) used by the CDN/CLI and some framework adapters. Not a public app-facing API.

### Related, not geo-policy

| Header | What it is |
| --- | --- |
| `x-nf-client-connection-ip` | Official client IP (Netlify also sets `x-forwarded-for`) |
| `x-nf-sign` | JWS on signed proxy rewrites |
| `x-nf-request-id` | Tracing |

## Proposed solution

Treat Netlify as a first-class geo provider, at least for:

1. The `x-country` country code
2. The official `context.geo` country + subdivision, wherever a Netlify `Context` is available

## Alternatives considered

- Document a workaround: copy `context.geo` onto a signal we already accept. That unblocks people today but is not a first-class integration.
- Parse `x-nf-geo`. More complete (includes region) but it is an undocumented internal header.

## Additional context

- [Netlify Functions API — `context.geo`](https://docs.netlify.com/build/functions/api/#geo)
- [Netlify Edge Functions API](https://docs.netlify.com/build/edge-functions/api/) — explicitly: no geo headers; use `context`

### PR DESCRIPTION
**Pull request by prashantbhudwal**
## Overview

Read Netlify's `x-country` header in `@c15t/backend` the same way Cloudflare and Vercel country headers are already read.

Netlify Functions send this header. Country only is enough. If an app needs a region, it can set `x-c15t-country` / `x-c15t-region` itself.

## Related Issue

Fixes #1023

## Type of Change

- [x] Enhancement (improves existing functionality)
- [x] Documentation

### Key Changes

1. Add `x-country` to the three existing country header lists.
2. Cover it with the same `getHeaders` tests used for `cf-ipcountry`.
3. Note it in the framework integration guide.

### Technical Notes

No new types, no second handler argument, no Hono bindings. Precedence is unchanged: `x-c15t-country`, then Cloudflare, Vercel, CloudFront, then `x-country`, then `x-country-code`.

## Testing

- [x] `x-country: DE` resolves as Germany
- [x] `cf-ipcountry` still wins over `x-country`
- [x] Edge `unstable_resolveConsent` uses the same header list

```sh
bun run --cwd packages/backend test src/handlers/init/index.test.ts src/handlers/init/geo.test.ts src/edge/resolve-consent.test.ts
bun turbo run test --filter=@c15t/backend
```

Focused tests: 3 files, 100 passed. Backend suite: 38 files, 570 passed.

### Edge Cases

- No UI changes.
- `x-country` is a platform header, same trust model as `cf-ipcountry`.

### Required

- [x] Issue is linked
- [x] Tests added/updated
- [x] Documentation updated
- [x] Tested locally
- [x] Code follows style guide
- [x] Commit messages follow [conventional commits](https://www.conventionalcommits.org/)

### CODE REVIEW

File: packages/backend/src/edge/init-handler.ts [hunk 1]
```diff
@@ -13,16 +13,27 @@ import {
...
  *
- * The returned function accepts a standard `Request` and returns a `Response`.
+ * The returned function accepts a standard `Request` and an optional request
+ * context. Netlify Edge Functions pass `context` as the second argument, which
+ * supplies country and subdivision via `context.geo`.
+ *
  * It has no dependency on Hono or any database adapter, making it suitable for
- * edge runtimes such as Vercel Middleware, Cloudflare Workers, or Deno Deploy.
+ * edge runtimes such as Vercel Middleware, Cloudflare Workers, Deno Deploy,
+ * or Netlify Edge Functions.
```
[the remark this answers is not in the document]
Comment (prashantbhudwal) [reply packages/backend/src/edge/init-handler.ts]: Added the missing `@param`, `@returns`, and `@throws` tags in 179ce704.

File: packages/backend/src/handlers/init/index.ts [hunk 2]
```diff
@@ -1,49 +1,23 @@
...
-		normalizeHeader(headers.get('x-region-code'));
-
-	// Get preferred language from Accept-Language header
-	const acceptLanguage = normalizeHeader(headers.get('accept-language'));
+export function getHeaders(
+	headers: Headers | undefined,
+	geo?: C15TGeoLocation | null
+) {
+	const { countryCode, regionCode } = extractLocation(headers, geo);
+	const acceptLanguage = headers?.get('accept-language') || null;
```
[the remark this answers is not in the document]
Comment (prashantbhudwal) [reply packages/backend/src/handlers/init/index.ts]: Corrected the return contract in 179ce704. It now states that unavailable fields are null.

File: packages/backend/src/handlers/init/geo.test.ts [hunk 3]
```diff
@@ -315,3 +316,176 @@ describe('checkJurisdiction', () => {
...
+	geo: {
+		country: { code: 'US' },
+		subdivision: { code: 'CA' },
+	},
+} satisfies C15TRequestContext;
+
+const netlifyQuebec = {
+	geo: {
+		country: { code: 'CA' },
+		subdivision: { code: 'CA-QC' },
```
[the remark this answers is not in the document]
Comment (prashantbhudwal) [reply packages/backend/src/handlers/init/geo.test.ts]: Updated the Netlify Quebec fixture to use the unprefixed `QC` code in 179ce704. The existing jurisdiction tests still cover `CA-QC` separately.

File: packages/backend/src/handlers/init/geo.ts [hunk 4]
```diff
@@ -14,20 +24,34 @@ function normalizeHeader(
...
-		normalizeHeader(headers.get('cf-ipcountry')) ??
-		normalizeHeader(headers.get('x-vercel-ip-country')) ??
-		normalizeHeader(headers.get('x-amz-cf-ipcountry')) ??
-		normalizeHeader(headers.get('x-country-code'));
+		normalizeHeader(headers?.get('x-c15t-country')) ??
+		normalizeHeader(geo?.country?.code) ??
+		normalizeHeader(headers?.get('cf-ipcountry')) ??
+		normalizeHeader(headers?.get('x-vercel-ip-country')) ??
+		normalizeHeader(headers?.get('x-amz-cf-ipcountry')) ??
+		normalizeHeader(headers?.get('x-country')) ??
```
[the remark this answers is not in the document]
Comment (prashantbhudwal) [reply packages/backend/src/handlers/init/geo.ts]: Added the trust note to the Netlify integration guide in 179ce704. It now warns that clients can supply `x-country` on other deployments.

**Review by KayleeWilliams — COMMENTED** [review]
Thanks for taking this on.

This should just be adding `x-country` to the existing country header list, next to `cf-ipcountry` / `x-vercel-ip-country`. Same pattern as the other providers. No new types, no second handler argument, no Hono bindings, no threading `geo` through the stack.

Netlify Functions already send that header. Country only is fine. If someone needs a region they can set `x-c15t-country` / `x-c15t-region` themselves, including from `context.geo` on Edge.

So: add `x-country`, a couple of tests, a sentence in the docs. The rest can go.

**Comment by prashantbhudwal** [comment]
Thanks for the the feedback @KayleeWilliams. I will get it done.
